lib/workload/stateless/stacks/filemanager/deploy/stack.ts

```diff
@@ -0,0 +1,129 @@
+
+export const FILEMANAGER_SERVICE_NAME = 'filemanager';
+
+/**
+ * Stateful config for filemanager.
+ */
+export type FilemanagerConfig = Omit<DatabaseProps, 'host' | 'securityGroup'> & {
+  eventSourceQueueName: string;
+  eventSourceBuckets: string[];
+  databaseClusterEndpointHostParameter: string;
+  vpcProps: VpcLookupOptions;
+  migrateDatabase?: boolean;
+  securityGroupName: string;
+  cognitoUserPoolIdParameterName: string;
+  cognitoPortalAppClientIdParameterName: string;
+  cognitoStatusPageAppClientIdParameterName: string;
+}
+
+/**
+ * Props for the filemanager stack.
+ */
+export type FilemanagerProps = StackProps & FilemanagerConfig;
+
+/**
+ * Construct used to configure the filemanager.
+ */
+export class Filemanager extends Stack {
+  private readonly vpc: IVpc;
+  private readonly host: string;
+  private readonly securityGroup: ISecurityGroup;
+  private readonly queue: IQueue;
+  
+  constructor(scope: Construct, id: string, props: FilemanagerProps) {
+    super(scope, id, props);
+
+    this.vpc = Vpc.fromLookup(this, 'MainVpc', props.vpcProps);
+
+    this.securityGroup = SecurityGroup.fromLookupByName(
+      this,
+      'OrcaBusLambdaSecurityGroup',
+      props.securityGroupName,
+      this.vpc
+    );
+
+    this.host = StringParameter.valueForStringParameter(
+      this,
+      props.databaseClusterEndpointHostParameter
+    );
+
+    if (props?.migrateDatabase) {
+      const migrateFunction = new MigrateFunction(this, 'MigrateFunction', {
+        vpc: this.vpc,
+        host: this.host,
+        port: props.port,
+        securityGroup: this.securityGroup,
+      });
+
+      new ProviderFunction(this, 'MigrateProviderFunction', {
+        vpc: this.vpc,
+        function: migrateFunction.function,
+      });
+    }
+
+    this.queue = Queue.fromQueueArn(
+      this,
+      'FilemanagerQueue',
+      Arn.format(
+        {
+          resource: props.eventSourceQueueName,
+          service: 'sqs',
+        },
+        this
+      )
+    );
+
+    this.createIngestFunction(props);
+    this.createQueryFunction(props);
+  }
+
+  /// Lambda function definitions and surrounding infra
+  // Ingest function
+  private createIngestFunction(props: FilemanagerProps) {
+    return new IngestFunction(this, 'IngestFunction', {
+      vpc: this.vpc,
+      host: this.host,
+      securityGroup: this.securityGroup,
+      eventSources: [this.queue],
+      buckets: props.eventSourceBuckets,
+      ...props
+    });
+  }
+
+  // Query function and API Gateway fronting the function
+  private createQueryFunction(props: FilemanagerProps) {
+    let objectsQueryLambda = new QueryFunction(this, 'ObjectsQueryFunction', {
+      vpc: this.vpc,
+      host: this.host,
+      securityGroup: this.securityGroup,
+      ...props
+    });
+
+    const ApiGateway = new ApiGatewayConstruct(this, 'ApiGateway-'+props.stackName, {
+      region: this.region,
+      ...props,
+    });
```

Comment:
For the construct ID, just 'ApiGateway' should do it. Then, we set the `apiName` prop like:

```
    const ApiGateway = new ApiGatewayConstruct(this, 'ApiGateway', {
      region: this.region,
      apiName: props.stackName
      ...props,
    });
```